src/Chainweb/Chainweb.hs

```diff
@@ -516,6 +529,20 @@ withChainwebInternal conf logger peer serviceSock rocksDb pactDbDir resetDb inne
             void $ _pactSyncToBlock pact bh
             logCr Info "pact db synchronized"
 
+    makeCheckpoint :: IO Text
+    makeCheckpoint = do
+        createDirectoryIfMissing False rocksDbCheckpointDir
+        Time (epochToNow :: TimeSpan Integer) <- getCurrentTimeIntegral
+        let time = microsToText (timeSpanToMicros epochToNow)
+        -- maxBound ~ always flush WAL log before checkpoint, under the assumption
+        -- that it's not much work
+        checkpointRocksDb rocksDb maxBound (rocksDbCheckpointDir </> T.unpack time)
+        return time
+
+    maybeMakeCheckpoint :: Maybe (IO Text)
+    maybeMakeCheckpoint = 
+        makeCheckpoint <$ guard (_configCheckpoints conf)
+
```

Comment:
I wasn't as big a fan of that path because it would require putting the `RocksDb` handle itself into as lot of places which try to abstract over the underlying database.

Reply:
I did something like that.